modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1116,6 +1116,17 @@ void ONNXImporter::populateNet(Net dstNet)
                 layerParams.set("zoom_factor_y", scales.getIntValue(2));
                 layerParams.set("zoom_factor_x", scales.getIntValue(3));
             }
+            else if(node_proto.input_size() == 2 && constBlobs.find(node_proto.input(1)) == constBlobs.end())
+            {
+                // upsample with two inputs
+                shapeIt = outShapes.find(node_proto.input(0));
+                MatShape inpShape1 = shapeIt->second;
+                shapeIt = outShapes.find(node_proto.input(1));
+                MatShape inpShape2 = shapeIt->second;
+                layerParams.set("zoom_factor_y", inpShape2[2]/inpShape1[2]);
+                layerParams.set("zoom_factor_x", inpShape2[3]/inpShape1[3]);
```

Comment:
Please check that number of dimensions equals 4 before getting it:
```
CV_CheckEQ(inpShape1.size(), 4, "Only resize for 4D tensors is supported.")
CV_CheckEQ(inpShape2.size(), 4, "Only resize for 4D tensors is supported.")
```

Reply:
Can you please fix it?

Reply:
changes are removed. resolving conversation.